SaveState: Keep an undo for each slot by default

unknownbrackets opened the pull request:
No real UI for it ATM, but at least there's a safety mechanism.  Otherwise you just find out you hit the wrong key or made a horrible mistake.

At least this way, you might find it or ask online.

-[Unknown]

unknownbrackets committed:
SaveState: Keep an undo for each slot by default.

unknownbrackets committed:
SaveState: Show undos in save data manager.

hrydgard commented:
It's a nice feature but I'm worried about unexpected disk utilization. A lot of people run their Android devices pretty full, and doubling the size of each save slot makes the impact pretty large, and without UI, it's not so easy for many of those to even make use of it. Maybe only default it to on for PC users?

unknownbrackets committed:
SaveState: Default undo slots off on mobile.

It's still a pretty useful feature on mobile (considering mistaps), but
it also eats a bunch of storage so it can always be turned back on.

unknownbrackets commented:
Right, makes sense.  I added to to the System menu so it could still be enabled easily on mobile.  More reason people might know it exists.

-[Unknown]

hrydgard commented on Core/SaveState.cpp:
```diff
@@ -433,12 +457,43 @@ namespace SaveState
 		}
 	}
 
+	bool UndoSaveSlot(const std::string &gameFilename, int slot) {
```
Nothing calls this function, I guess it's here for future UI?

unknownbrackets replied:
Right, I just had trouble putting a button in a good place that didn't seem ugly or annoying.

-[Unknown]
